src/wayland/shortcuts_inhibit/inhibitor.cpp

```diff
@@ -0,0 +1,180 @@
+#include "inhibitor.hpp"
+
+#include <private/qwaylandwindow_p.h>
+#include <qlogging.h>
+#include <qobject.h>
+#include <qtmetamacros.h>
+
+#include "../../window/proxywindow.hpp"
+#include "../../window/windowinterface.hpp"
+#include "proto.hpp"
+
+namespace qs::wayland::shortcuts_inhibit {
+using QtWaylandClient::QWaylandWindow;
+
+ShortcutsInhibitor::ShortcutsInhibitor() {
+	this->bBoundWindow.setBinding([this] {
+		return this->bEnabled ? this->bWindowObject.value() : nullptr;
+	});
+}
+
+ShortcutsInhibitor::~ShortcutsInhibitor() {
+	if (!this->inhibitor) return;
+
+	auto* manager = impl::ShortcutsInhibitManager::instance();
+	if (!manager) return;
+
+	QObject::disconnect(this->inhibitor, nullptr, this, nullptr);
+	manager->unrefShortcutsInhibitor(this->inhibitor);
+}
+
+QObject* ShortcutsInhibitor::window() const { return this->bWindowObject; }
+
+void ShortcutsInhibitor::setWindow(QObject* window) {
+	if (window == this->bWindowObject) return;
+
+	auto* proxyWindow = qobject_cast<ProxyWindowBase*>(window);
+
+	if (proxyWindow == nullptr) {
+		if (auto* iface = qobject_cast<WindowInterface*>(window)) {
+			proxyWindow = iface->proxyWindow();
+		}
+	}
+
+	this->bWindowObject = proxyWindow ? window : nullptr;
+}
+
+bool ShortcutsInhibitor::isActive() const {
+	return this->inhibitor ? this->inhibitor->isActive() : false;
+}
+
+void ShortcutsInhibitor::boundWindowChanged() {
+	auto* window = this->bBoundWindow.value();
+	auto* proxyWindow = qobject_cast<ProxyWindowBase*>(window);
+
+	if (proxyWindow == nullptr) {
+		if (auto* iface = qobject_cast<WindowInterface*>(window)) {
+			proxyWindow = iface->proxyWindow();
+		}
+	}
```

Comment:
If we set bWindowObject, and this triggers boundWindowChanged, proxyWindow should not be set in setWindow but instead only the binding. Additionally, bindable properties internally diff state so you dont need the == check if you're just setting a bindable